services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/ProjectSettingsManagerImpl.java

```diff
@@ -189,25 +195,33 @@ public void deleteProjectSetting(UserInfo userInfo, String id) throws DatastoreE
 			//A valid bucket name must also be a valid domain name
 			ValidateArgument.requirement(InternetDomainName.isValid(externalS3StorageLocationSetting.getBucket()), "Invalid Bucket Name");
 
-			validateBucketAccess(externalS3StorageLocationSetting);
-			validateOwnership(externalS3StorageLocationSetting, userProfile);
+			validateS3BucketAccess(externalS3StorageLocationSetting);
+			validateS3BucketOwnership(externalS3StorageLocationSetting, userProfile);
+		} else if (storageLocationSetting instanceof ExternalGoogleCloudStorageLocationSetting) {
+			UserProfile userProfile = userProfileManager.getUserProfile(userInfo.getId().toString());
```

Comment:
getusername not the whole thing. (principal alias dao perhaps)

Reply:
We also allow users to enter an email address in `owner.txt`. Still need the user's full profile, some refactoring will be done to attempt to simplify